public/app/core/components/Select/DashboardPicker.tsx

```diff
@@ -56,19 +58,39 @@ export const DashboardPicker = ({
     (async () => {
       // value was manually changed from outside or we are rendering for the first time.
       // We need to fetch dashboard information.
-      const res = await getDashboardAPI().getDashboardDTO(value, undefined);
-      const formatted = ResponseTransformers.ensureV1Response(res);
+      // when using getDashboardAPI, if isV2Mode is not set, we will always return the v1 schema version
+      // isV2Mode is set to true when the feature toggle of dashboardSchemaV2 and useV2DashboardsAPI is enabled
+      // We could create a different feature toggle with better naming if we want, but the idea of using
+      // dashboardSchemaV2 is to indicate that this component is ready to process the new schema
+      const isUIReadyForV2 = config.featureToggles.useV2DashboardsAPI;
+
+      if (isUIReadyForV2) {
+        const resWithSchemaV2 = await getDashboardAPI({ useV2Mode: true }).getDashboardDTO(value, undefined);
```

Comment:
can we make `getDashboardAPI({ useV2Mode: true })` simply `getDashboardAPI('v2')`